libcxx/include/__expected/expected.h

```diff
@@ -955,8 +956,17 @@ class expected {
     _LIBCPP_NO_UNIQUE_ADDRESS _ErrorType __unex_;
   };
 
+  _LIBCPP_HIDE_FROM_ABI static constexpr auto __calculate_padding() {
```

Comment:
This won't work if there is no tail padding, since we'll end up creating a `char __padding_[0];` below. This is the case for e.g. `std::expected<char, char>`. Needs tests.

Reply:
I implemented an alternative approach to calculate the padding type that avoids zero-length arrays (still wondering if there is a more elegant way, though). I also added some tests.